src/renderer/components/badge/badge.tsx

```diff
@@ -13,6 +13,15 @@ export interface BadgeProps extends React.HTMLAttributes<any>, TooltipDecoratorP
   isExpanded?: boolean; // always force state to this value
 }
 
+const badgeMeta = observable({
+  hasTextSelected: false,
+});
+
+// Common handler for all Badge instances
+document.addEventListener("selectionchange", () => {
+  badgeMeta.hasTextSelected = window.getSelection().toString().trim().length > 0
+});
```

Comment:
This is no different than checking it in the handler because clearing the selection will clear this field too. That is why I was adding the `||=`